_inc/client/components/dash-section-header/style.scss

```diff
@@ -45,6 +55,12 @@
 	@include breakpoint( "<660px" ) {
 		margin-top: rem( 4px );
 	}
+
+	@include breakpoint( "<480px" ) {
+		display: block;
+		width: 100%;
+		margin-top: rem( 32px );
```

Comment:
There aren't any floats. But I figured it out. Just removed the flexbox of the parent on small screens. Good to go. Top margin is gone.